sdk-utils/src/main/kotlin/com/rakuten/tech/mobile/sdkutils/eventlogger/EventLogger.kt

```diff
@@ -1,28 +1,62 @@
 package com.rakuten.tech.mobile.sdkutils.eventlogger
 
 import android.content.Context
+import androidx.annotation.VisibleForTesting
 import com.rakuten.tech.mobile.sdkutils.BuildConfig
 import com.rakuten.tech.mobile.sdkutils.logger.Logger
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import java.lang.ref.WeakReference
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
 
 /**
  * A remote logging utility that sends events to the Event Logger platform.
  * This is intended to be used internally by Rakuten's SDKs.
  */
 @SuppressWarnings(
-    "EmptyFunctionBlock"
+    "EmptyFunctionBlock",
+    "TooManyFunctions",
+    "LargeClass"
```

Comment:
Can we adopt `Has-A` practice to avoid `LargeClass` and `TooManyFunctions` ?

- I mean to have some specific set of methods in a separate class/Interface and then using instance of that class/Interface to call the main method.
- and we may choose to initialise those instance at init.

Reply:
That is a good idea, but as this is a utility SDK and does not expect to be a full SDK with more and more functionalities, I'd like to make it as simple as possible and consistent with other packages. Right now some components are already moved to a separate class, we can make moving other functionality as tech debt to solve these suppressions.